W-13480276 promoting-api-instance-kl.adoc (#538)
mulesoft/docs-api-manager: modules/ROOT/pages/promote-api-task.adoc

include::_attributes.adoc[]
endif::[]

> Before promoting an API instance to another environment:

Reviewer: See google doc:

https://docs.google.com/document/d/1_CNwMZhAFZGxBLBNo2VRuKZnmPGAy2-ikd-P5YUrgVE/edit?usp=sharing

. Have the `Manage APIs Configuration` permission to promote an API.
. Have the `Access Management` permission to set permissions for viewing and managing an entity.

You can only promote an entity (Policies, Alerts, and more) if you have both permission to view the source and to manage the destination over each entity. Typical view permissions include `View API Configuration` and `View Policies`, as explained in xref:environment-permission-task.adoc[To Grant Permission to Access Environments]. 

Set the permissions as needed before proceeding to the following steps:

include::partial$task-promote-api.adoc[]

> Promoting an API affects only the API definition, not the existing deployed applications subscribed to the API. After promoting an API, the API Name remains the same, but the API ID changes.

Reviewer: ```suggestion
Promoting an API instance affects only the API definition, not the existing deployed applications subscribed to the instance. After promoting an API, the *API Label* remains the same, but the *API Instance ID* changes.
```

> For existing applications that need to subscribe to a promoted API:

Reviewer: ```suggestion
For existing applications that must subscribe to a promoted API instance:
```

> . Update the API ID.

Reviewer: ```suggestion
. Update the *API Instance ID* within the application.
```

include::_attributes.adoc[]
endif::[]

> Before promoting an API instance to another environment ensure that you have the `Manage APIs Configuration` permission.

Reviewer: ```suggestion
Before promoting an API instance to another environment, ensure that you have the `Manage APIs Configuration` permission.
```

> Before promoting an API instance to another environment, ensure that you have the `Manage APIs Configuration` permission.

Reviewer 2: I don't think we change fonts for permissions.

Reviewer 2: https://confluence.internal.salesforce.com/display/MTDT/MuleSoft+CX+Writing+Style+Reference#MuleSoftCXWritingStyleReference-fontchoices

> You can only promote an entity, such as policies or alerts, if you have both the permissions to view the entity and to manage its destination. For example, you must have the `View Policies` and `Manage Policies` permissions to promote a policy. 

Reviewer 2: font

To promote an API instance:

include::partial$task-promote-api.adoc[]

Promoting an API instance affects only the API definition, not the existing deployed applications subscribed to the instance. After promoting an API, the *API Label* remains the same, but the *API Instance ID* changes.

For existing applications that must subscribe to a promoted API instance:

. Update the *API Instance ID* within the application.

> . Because each environment has a different client ID and secret, update the application's client ID and secret.

Reviewer 2: ```suggestion
. Update the application's client ID and secret.
+
Each environment has a different client ID and secret.
```
